esphome/components/grove_i2c_motor/__init__.py

```diff
@@ -0,0 +1,152 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import automation
+from esphome.components import i2c
+
+from esphome.const import (
+    CONF_ID,
+    CONF_CHANNEL,
+    CONF_SPEED,
+    CONF_DIRECTION,
+)
+
+DEPENDENCIES = ["i2c"]
+
+CODEOWNERS = ["@max246"]
+
+grove_i2c_motor_ns = cg.esphome_ns.namespace("grove_i2c_motor")
+GROVE_TB6612FNG = grove_i2c_motor_ns.class_(
+    "GroveMotorDriveTB6612FNG", cg.Component, i2c.I2CDevice
+)
+GROVETB6612FNGMotorRunAction = grove_i2c_motor_ns.class_(
+    "GROVETB6612FNGMotorRunAction", automation.Action
+)
+GROVETB6612FNGMotorBrakeAction = grove_i2c_motor_ns.class_(
+    "GROVETB6612FNGMotorBrakeAction", automation.Action
+)
+GROVETB6612FNGMotorStopAction = grove_i2c_motor_ns.class_(
+    "GROVETB6612FNGMotorStopAction", automation.Action
+)
+GROVETB6612FNGMotorStandbyAction = grove_i2c_motor_ns.class_(
+    "GROVETB6612FNGMotorStandbyAction", automation.Action
+)
+GROVETB6612FNGMotorNoStandbyAction = grove_i2c_motor_ns.class_(
+    "GROVETB6612FNGMotorNoStandbyAction", automation.Action
+)
+
+DIRECTION_TYPE = {
+    "FORWARD": 1,
+    "BACKWARD": 2,
+}
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.declare_id(GROVE_TB6612FNG),
+        }
+    )
+    .extend(cv.COMPONENT_SCHEMA)
+    .extend(i2c.i2c_device_schema(0x14))
+)
+
+
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await i2c.register_i2c_device(var, config)
+
+
+@automation.register_action(
+    "grove_i2c_motor.motor_run",
+    GROVETB6612FNGMotorRunAction,
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.use_id(GROVE_TB6612FNG),
+            cv.Required(CONF_CHANNEL): cv.int_range(min=0, max=1),
+            cv.Required(CONF_SPEED): cv.int_range(min=0, max=255),
```

Comment:
You access these as templatable below, but they are not in the schema? Same for other actions below.
```suggestion
            cv.Required(CONF_CHANNEL): cv.templatable(cv.int_range(min=0, max=1)),
            cv.Required(CONF_SPEED): cv.templatable(cv.int_range(min=0, max=255)),
```